Bootstrapper/appleGuiceBootstrapper.cpp

```diff
@@ -341,43 +346,63 @@ string readFromStdin() {
     return stringBuilder.str();
 }
 
-unordered_map<string, set<string> > generateProtocolsToImpsMap(string headerEntriesAsString) {
-    unordered_map<string, set<string> > protocolToSuperProtocols;
-    unordered_map<string, set<string> > classToProtocols;
-    unordered_map<string, string> classToSuperClass;
-    
-    unordered_map<string, set<string> > protocolToImps;
-    
-    vector<string> headerEntries = split(headerEntriesAsString, "\n");
-    
-    //parse
-    for(vector<string>::const_iterator headerEntriesIterator = headerEntries.begin(); headerEntriesIterator != headerEntries.end(); headerEntriesIterator++ ) {
-        string entry = *headerEntriesIterator;
-        entry = trim(entry);
-        parseLine(entry, protocolToSuperProtocols, classToSuperClass, classToProtocols);
-    }
+unordered_map<string, set<string> > generateProtocolsToImpsMap(string headerEntriesAsString, set<string> prefixesToIgnore) {
+	unordered_map<string, set<string> > protocolToSuperProtocols;
+	unordered_map<string, set<string> > classToProtocols;
+	unordered_map<string, string> classToSuperClass;
+
+	unordered_map<string, set<string> > protocolToImps;
     
-    //build dictionary
+  	vector<string> headerEntries = split(headerEntriesAsString, "\n");
+
+  	//parse
+  	for(vector<string>::const_iterator headerEntriesIterator = headerEntries.begin(); headerEntriesIterator != headerEntries.end(); headerEntriesIterator++ ) {
+		string entry = *headerEntriesIterator;
+		entry = trim(entry);
+		parseLine(entry, protocolToSuperProtocols, classToSuperClass, classToProtocols);
+  	}
+
+  	//build dictionary
     for (unordered_map<string, string>::const_iterator it = classToSuperClass.begin(); it != classToSuperClass.end(); ++it) {
-        string className = it->first;
-        travereOverClassHierarchAndAct(className, protocolToSuperProtocols, classToSuperClass, classToProtocols, protocolToImps);
+    	string className = it->first;
+
+        set<string> implementedProtocols;
+        
+        protocolsImplementedByClass(className, classToSuperClass, classToProtocols, prefixesToIgnore, implementedProtocols);
+        
+        addBindToResultList(className, implementedProtocols, protocolToSuperProtocols, protocolToImps);
     }
     return protocolToImps;
 }
 
+set<string> getPrefixesToIgnore(string prefixesStr) {
+    set<string> prefixesToIgnore;
+    vector<string> prefixEntries = split(prefixesStr, ",");
+    
+    for(vector<string>::const_iterator prefixesIterator = prefixEntries.begin(); prefixesIterator != prefixEntries.end(); prefixesIterator++ ) {
+        string prefix = *prefixesIterator;
+        prefixesToIgnore.insert(trim(prefix));
```

Comment:
prefixesToIgnore simply saves time by excluding unnecessary class names from checking. 
All your tests are on the result of the bootstrapper. 
You want me to write tests for a specific c++ method?

Reply:
Add a  test(s) in `BootstrapperTests.m` where a value is sent in arg 1 and as a result,  a class is being excluded from the output.
These tests are blackbox tests, they test only the input and output, not a specific c++ method.

Reply:
It wouldn't appear in the output without my changes. All it does is prevent traversal on UIView and UIViewController, because we can be sure that UIView is not implementing any protocol we define, or inherits from a class that we define that implements a certain protocol.

Reply:
You can write a test that verifies that ignored prefixes are not part of the output.

Reply:
Ok added tests